libbpf-tools/mountsnoop.c

```diff
@@ -289,11 +308,15 @@ int main(int argc, char **argv)
 		goto cleanup;
 	}
 
-	pb = perf_buffer__new(bpf_map__fd(obj->maps.events), PERF_BUFFER_PAGES,
-			      handle_event, handle_lost_events, NULL, NULL);
-	if (!pb) {
+	if (use_ringbuf)
+		buf = bpf_buffer__new(obj->maps.events_ringbuf,
+				      handle_ringbuf_event, NULL, NULL, NULL);
+	else
+		buf = bpf_buffer__new(obj->maps.events_perfbuf,
+				      handle_perfbuf_event, handle_lost_events, NULL, NULL);
```

Comment:
similarly, why these differences, just make bpf_buffer__new() accept unified callback (no cpu arg, decide on int/void return result) and optional lost_events (which won't be ever called for ringbuf, unless you add atomic counters in compat.bpf.c, which might be not too bad of an idea). Handle such differences inside the compat code, basically

Reply:
This should be done after skeleton load, so I added a new API for it.